cmd/kava/cmd/rocksdb/compact.go

```diff
@@ -0,0 +1,170 @@
+//go:build rocksdb
+// +build rocksdb
+
+package rocksdb
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"strconv"
+	"syscall"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/server"
+	"github.com/linxGnu/grocksdb"
+	"github.com/spf13/cobra"
+
+	"github.com/tendermint/tendermint/libs/log"
+)
+
+var CompactRocksDBCmd = &cobra.Command{
+	Use:   "compact <state|blockstore>",
+	Short: "force compacts RocksDB",
+	Long: `This is a utility command that performs a force compaction on the state or
+blockstore. This should only be run once the node has stopped.`,
+	Args: cobra.ExactArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		logger := log.NewTMLogger(log.NewSyncWriter(os.Stdout))
+
+		clientCtx := client.GetClientContextFromCmd(cmd)
+		ctx := server.GetServerContextFromCmd(cmd)
+
+		if server.GetAppDBBackend(ctx.Viper) != "rocksdb" {
+			return errors.New("compaction is currently only supported with rocksdb")
+		}
+
+		if args[0] != "state" && args[0] != "blockstore" {
+			return errors.New("invalid db name, must be either 'state' or 'blockstore'")
+		}
+
+		compactRocksDBs(clientCtx.HomeDir, logger, args[0])
+		return nil
+	},
+}
+
+// compactRocksDBs performs a manual compaction on the given db.
+func compactRocksDBs(
+	rootDir string,
+	logger log.Logger,
+	dbName string,
+) {
+	options := grocksdb.NewDefaultOptions()
+
+	dbPath := filepath.Join(rootDir, "data", dbName+".db")
+	logger.Info("opening db", "path", dbPath)
+
+	store, err := grocksdb.OpenDb(options, dbPath)
+	if err != nil {
+		logger.Error("failed to initialize cometbft db", "path", dbPath, "err", err)
+		return
+	}
+	defer store.Close()
+
+	logColumnFamilyMetadata(store, logger)
+
+	logger.Info("starting compaction...", "db", dbPath)
+
+	done := make(chan bool, 1)
+	registerSignalHandler(store, logger, done)
+	startCompactionStatsOutput(store, logger, done)
+
+	// Actually run the compaction
+	store.CompactRange(grocksdb.Range{Start: nil, Limit: nil})
```

Comment:
Just a nit, but lets name this `db` or something similar.  For me `store` always makes me thinkn of the cosmos-sdk store, not a raw rocksdb struct